Bio/Align/__init__.py

```diff
@@ -926,6 +928,429 @@ def sort(self, key=None, reverse=False):
+        for end1, end2 in self.path[1:]:
+            count1 = end1 - start1
+            count2 = end2 - start2
+            if count1 == 0:
+                if start2 == 0:
+                    Qstart += count2
+                elif end2 == n2:
+                    Qend -= count2
+                else:
+                    Qgapcount += 1
+                    Qgapbases += count2
+                start2 = end2
+            elif count2 == 0:
+                if start1 == 0:
+                    Tstart += count1
+                elif end1 == n1:
+                    Tend -= count1
+                else:
+                    Tgapcount += 1
+                    Tgapbases += count1
+                start1 = end1
+            else:
+                assert count1 == count2
+                tStarts.append(start1)
+                qStarts.append(start2)
+                blockSizes.append(count1)
+                for c1, c2 in zip(seq1[start1:end1], seq2[start2:end2]):
+                    if c1 == 'N' or c2 == 'N':
+                        Ns += 1
+                    elif c1 == c2:
+                        match += 1
+                    else:
+                        mismatch += 1
+                start1 = end1
+                start2 = end2
+        blockcount = len(blockSizes)
+        blockSizes = ",".join(map(str, blockSizes)) + ","
+        qStarts = ",".join(map(str, qStarts)) + ","
+        tStarts = ",".join(map(str, tStarts)) + ","
+        words = [str(match),
+                 str(mismatch),
+                 str(repmatch),
+                 str(Ns),
+                 str(Qgapcount),
+                 str(Qgapbases),
+                 str(Tgapcount),
+                 str(Tgapbases),
+                 strand,
+                 Qname,
+                 str(Qsize),
+                 str(Qstart),
+                 str(Qend),
+                 Tname,
+                 str(Tsize),
+                 str(Tstart),
+                 str(Tend),
+                 str(blockcount),
+                 blockSizes,
+                 qStarts,
+                 tStarts,
+                 ]
+        line = "\t".join(words) + "\n"
+        return line
+
+
+class Alignments(object):
+    """Implements an iterator over pairwise alignments returned by the aligner.
+
+    This class also supports indexing, which is fast for increasing indices,
+    but may be slow for random access of a large number of alignments.
+
+    Note that pairwise aligners can return an astronomical number of alignments,
+    even for relatively short sequences, if they align poorly to each other. We
+    therefore recommend to first check the number of alignments, accessible as
+    len(alignments), which can be calculated quickly even if the number of
+    alignments is very large.
+    """
+
+    def __init__(self, seqA, seqB, score, paths):
+        """Initialize a new Alignments object.
+
+        Arguments:
+         - seqA  - The first sequence, as a plain string, without gaps.
```

Comment:
Can this cope with ``seqA`` and/or ``seqB`` as ``Seq`` objects?

If not why not? If yes, we should say so in the docstring.

Reply:
Typically users wouldn't instantiate an object of this class. Alignments objects are created and returned by Aligner.align. This docstring is included only for completeness (and because Travis was complaining about it).

Reply:
I'm in favour of having the docstring and including this information, but fair point - given it is not expected to be initialised by the user, my query is not that important.